module/zcommon/zfs_uio.c

```diff
@@ -98,6 +106,50 @@ uiomove(void *p, size_t n, enum uio_rw rw, struct uio *uio)
 	}
 	return (0);
 }
+
+static int
+uiomove_bvec(void *p, size_t n, enum uio_rw rw, struct uio *uio)
+{
+	const struct bio_vec *bv = uio->uio_bvec;
+	size_t skip = uio->uio_skip;
+	ulong_t cnt;
+
+	ASSERT3U(skip, <, bv->bv_len);
+
+	while (n && uio->uio_resid) {
+		void *paddr;
+		cnt = MIN(bv->bv_len - skip, n);
+
+		paddr = zfs_kmap_atomic(bv->bv_page, KM_USER1);
```

Comment:
@ryao 
http://lxr.free-electrons.com/source/include/linux/uaccess.h#L16
```
  7 /*
  8  * These routines enable/disable the pagefault handler in that
  9  * it will not take any locks and go straight to the fixup table.
 10  *
 11  * They have great resemblance to the preempt_disable/enable calls
 12  * and in fact they are identical; this is because currently there is
 13  * no other way to make the pagefault handlers do this. So we do
 14  * disable preemption but we don't necessarily care about that.
 15  */
 16 static inline void pagefault_disable(void)
```